### PR TITLE
Invalid char input

### LINKED ISSUE #1188: Incorrect behaviour of CTRL+Backspace hotkey in TextBox
Tested on windows 10 with avalonia 0.5.0 and 0.5.2-build3963-alpha on core project created by Avalonia for Visual Studio extension.

9ff2b3ab-3def-403a-803b-dc5b1242212e

[AvaloniaCoreBackspaceProblemApp.zip](https://github.com/AvaloniaUI/Avalonia/files/1355996/AvaloniaCoreBackspaceProblemApp.zip)

### PR DESCRIPTION
**Pull request by JurjenBiewenga**
Added an array that is used to filter out problematic characters like ''.
This fixes #1188

### CODE REVIEW

File: src/Avalonia.Input/KeyboardDevice.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ public class KeyboardDevice : IKeyboardDevice, INotifyPropertyChanged
         public IInputManager InputManager => AvaloniaLocator.Current.GetService<IInputManager>();
 
         public IFocusManager FocusManager => AvaloniaLocator.Current.GetService<IFocusManager>();
+        
+        public string[] invalidCharacters = new String[1]{""};
```
Comment (kekekeks) [inline src/Avalonia.Input/KeyboardDevice.cs]: Do we need that `[1]` here? Also, why isn't it a `char` array? Are there any unicode surrogate pairs that we should care of?

File: src/Avalonia.Input/KeyboardDevice.cs [hunk 2]
```diff
@@ -122,18 +124,32 @@ public void ProcessRawEvent(RawInputEventArgs e)
...
+                        element.RaiseEvent(ev);
+                        e.Handled = ev.Handled;
+                    }
                 }
             }
         }
+
+        public string RemoveInvalidCharacters(string text)
+        {
+            for (var i = 0; i < invalidCharacters.Length; i++)
```
Comment (kekekeks) [inline src/Avalonia.Input/KeyboardDevice.cs]: Code is bloody inefficient, but I don't know if we care...

File: src/Avalonia.Input/KeyboardDevice.cs (hunk 1 above)
Comment (JurjenBiewenga) [reply src/Avalonia.Input/KeyboardDevice.cs]: You can't string.replace with an empty char.

File: src/Avalonia.Input/KeyboardDevice.cs (hunk 2 above)
Comment (JurjenBiewenga) [reply src/Avalonia.Input/KeyboardDevice.cs]: Feel free to suggest a better way

**Review by grokys — CHANGES_REQUESTED** [review]
I don't think this is the right approach - we try to follow WPF/UWP's example where possible and this character isn't filtered out in those frameworks. This probably should be done at the `TextBox` level rather than at the `KeyboardDevice` level.

File: src/Avalonia.Controls/TextBox.cs [hunk 3]
```diff
@@ -97,6 +97,7 @@ public UndoRedoState(string text, int caretPosition)
         private UndoRedoHelper<UndoRedoState> _undoRedoHelper;
         private bool _ignoreTextChanges;
         private IEnumerable<Exception> _dataValidationErrors;
+        private string[] invalidCharacters = new String[1]{""};
```
Comment (grokys) [inline src/Avalonia.Controls/TextBox.cs]: This looks like an empty string in the VS editor. Could you change it to read "\u007f" so it's more clear what's happening?

Also this doesn't need to be an instance member - it should be `static readonly`.

**Review by grokys — CHANGES_REQUESTED** [review]
Yes, I think this is a better way to do it, a few small changes still required though. Also a unit test would be useful!

**Review by grokys — APPROVED** [review]

**Pull request merged**